mathlibtools/lib.py

```diff
@@ -117,14 +117,19 @@ def download(url: str, target: Path) -> None:
     total_size = int(req.headers.get('content-length', 0))
     BLOCK_SIZE = 1024
     progress = tqdm(total=total_size, unit='iB', unit_scale=True)
-    with target.open('wb') as tgt:
-        for data in req.iter_content(BLOCK_SIZE):
-            progress.update(len(data))
-            tgt.write(data)
+    for data in req.iter_content(BLOCK_SIZE):
+        progress.update(len(data))
+        tgt.write(data)
     progress.close()
     if total_size != 0 and progress.n != total_size:
         raise LeanDownloadError('Failed to download ' + url)
 
+def download(url: str, target: Path) -> None:
+    """Download from url into the target path"""
+    with tempfile.NamedTemporaryFile(mode='wb') as temp:
+        log.info('Trying to download {} to {} (temporary file: {})'.format(url, target, temp.name))
+        download_to_file(url, temp)
+        shutil.copy(temp.name, target)
```

Comment:
```suggestion
        os.rename(temp.name, target)
```
`shutil.copy` is not atomic.  You'll also need to create the temporary file in the same directory as target ofc.

Reply:
Looking into it some more, doing this completely correctly across all OSes is not trivial at all. So I just went with a library instead.